doc: add suggested tsconfig for type stripping (#55534)
nodejs/node: doc/api/typescript.md

generation, and by replacing inline types with whitespace, Node.js can run
TypeScript code without the need for source maps.

The suggested `tsconfig.json` settings for type stripping are:

```json
{
    "compilerOptions": {

>         "allowImportingTsExtensions": true,

Reviewer: ```suggestion
        "module": "nodenext",
        "allowImportingTsExtensions": true,
```

Actually, this may be a good idea.

Reviewer: See: https://www.typescriptlang.org/docs/handbook/modules/guides/choosing-compiler-options.html#im-compiling-and-running-the-outputs-in-nodejs (roughly)

        "rewriteRelativeImportExtensions": true,

>         "verbatimModuleSyntax": true,

Reviewer: You all would know better than me, but does whatever renders this markdown care about the trailing comma in a `json` code fence?

Author: fixed anyways

generation, and by replacing inline types with whitespace, Node.js can run
TypeScript code without the need for source maps.

> The suggested `tsconfig.json` settings for type stripping are:

Reviewer 2: ```suggestion
Type stripping works with most versions of TypeScript but we recommend version 5.7 or newer with the following `tsconfig.json` settings:
```

```json
{
  "compilerOptions": {

>      "module": "nodenext",

Reviewer 3: ```suggestion
     "target": "esnext", 
     "module": "nodenext",
```

Amaro does not perform down-levelling (compiling new JS features into older JS). This ensures the TS output matches that expectation.

Reviewer 2: I wish TS would change the default behavior to target esnext. Maybe one day.

     "allowImportingTsExtensions": true,

>      "rewriteRelativeImportExtensions": true,

Reviewer 2: I'm [not seeing any documentation](https://www.typescriptlang.org/tsconfig/) for `rewriteRelativeImportExtensions`.

Which version of TS was this introduced in?

And should we mention the minimum version of TS in the docs here?

Reviewer: It's not out yet; due in 5.7: https://devblogs.microsoft.com/typescript/announcing-typescript-5-7-beta/#path-rewriting-for-relative-paths

Reviewer 2: I see. I made a suggestion to mention 5.7 or newer above 👍

generation, and by replacing inline types with whitespace, Node.js can run
TypeScript code without the need for source maps.

Type stripping works with most versions of TypeScript

> but we recommend version 5.7 or newer with the following `tsconfig.json` settings:

Reviewer 4: ```suggestion
but recommendation is version 5.7 or newer with the following `tsconfig.json` settings:
```
